packages/chat-stateful-client/src/ResourceDownloadQueue.ts

```diff
@@ -188,7 +194,12 @@ export const fetchImageSource = async (
     clearTimeout(id);
     return response;
   }
-  const accessToken = await credential.getToken();
+
+  const accessToken = await authentication.credential.getToken();
+  if (!src.startsWith(authentication.endpoint)) {
```

Comment:
This must not be `startsWith`.
We need to check that the root of the domain of the URL is `communication.azure.com`
We must also verify that the protocol is `https`